alpha/declcfg/declcfg_to_model.go

```diff
@@ -39,67 +86,47 @@ func ConvertToModel(cfg DeclarativeConfig) (model.Model, error) {
 			return nil, fmt.Errorf("unknown package %q for bundle %q", b.Package, b.Name)
 		}
 
-		props, err := parseProperties(b.Properties)
+		props, err := property.Parse(b.Properties)
 		if err != nil {
 			return nil, fmt.Errorf("parse properties for bundle %q: %v", b.Name, err)
 		}
 
-		if len(props.Packages) == 0 {
-			return nil, fmt.Errorf("missing package property for bundle %q", b.Name)
+		if len(props.Packages) != 1 {
+			return nil, fmt.Errorf("package %q bundle %q must have exactly 1 %q property, found %d", b.Package, b.Name, property.TypePackage, len(props.Packages))
 		}
 
 		if b.Package != props.Packages[0].PackageName {
 			return nil, fmt.Errorf("package %q does not match %q property %q", b.Package, property.TypePackage, props.Packages[0].PackageName)
 		}
 
-		if len(props.Channels) == 0 {
-			return nil, fmt.Errorf("bundle %q is missing channel information", b.Name)
+		// Parse version from the package property.
+		ver, err := semver.Parse(props.Packages[0].Version)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing bundle version: %v", err)
 		}
 
-		for _, bundleChannel := range props.Channels {
-			pkgChannel, ok := mpkg.Channels[bundleChannel.Name]
-			if !ok {
-				pkgChannel = &model.Channel{
-					Package: mpkg,
-					Name:    bundleChannel.Name,
-					Bundles: map[string]*model.Bundle{},
-				}
-				if bundleChannel.Name == defaultChannelName {
-					mpkg.DefaultChannel = pkgChannel
-				}
-				mpkg.Channels[bundleChannel.Name] = pkgChannel
-			}
-
-			// Parse version from the package property, falling back to the CSV's spec.version field.
```

Comment:
@estroz Was this introduced with the `opm diff` stuff? What was the purpose of falling back to the CSV `spec.version` field? The `olm.package` property is required to have a valid semver version in it for all bundles.

Reply:
Oh perfect. In that case fallback is unnecessary.